tools/stylelint/selector-no-deep/index.js

```diff
@@ -0,0 +1,37 @@
+const stylelint = require('stylelint');
+const isStandardSyntaxRule = require('stylelint/lib/utils/isStandardSyntaxRule');
+const isStandardSyntaxSelector = require('stylelint/lib/utils/isStandardSyntaxSelector');
+
+const ruleName = 'material/selector-no-deep';
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  expected: selector => `Usage of the /deep/ in "${selector}" is now allowed`,
+});
+
+
+/**
+ * Stylelint plugin that prevents uses of /deep/ in selectors.
+ */
+const plugin = stylelint.createPlugin(ruleName, isEnabled => {
+  return (root, result) => {
+    if (!isEnabled) return;
+
+    root.walkRules(rule => {
+      if (rule.parent.type === 'rule' &&
+          isStandardSyntaxRule(rule) &&
+          isStandardSyntaxSelector(rule.selector) &&
+          rule.selector.indexOf('/deep/') > -1) {

```

Comment:
I'm not sure what the support for that one is in Node. It's not terribly good in browsers.